File: 104 maxDepth/104.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        queue,res = [root],0
        while queue:
            tmp=[]
            for node in queue:
                if node.left:
                    tmp.append(node.left)
                if node.right:
                    tmp.append(node.right)
            queue=tmp[:]
            res+=1
        return res

File: 104 maxDepth/104.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

int maxDepth(struct TreeNode* root){
    return getDepth(root);
}

int getDepth(struct TreeNode* root){
    if(root==NULL){
        return 0;
    }
    int ld=getDepth(root->left);
    int rd=getDepth(root->right);
    return 1+(ld>rd?ld:rd);
}